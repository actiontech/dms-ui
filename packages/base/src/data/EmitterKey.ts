enum EmitterKey {
  DMS_Refresh_User_List = 'DMS_REFRESH_USER_LIST',
  DMS_Refresh_Role_List = 'DMS_REFRESH_ROLE_LIST',
  DMS_Refresh_User_Group_List = 'DMS_REFRESH_USER_GROUP_LIST',
  DMS_Refresh_User_Permission_list = 'DMS_Refresh_User_Permission_list',
  Reset_Test_Data_Source_Connect = 'DMS_Reset_Test_Data_Source_Connect',
  DMS_Reset_DataSource_Form = 'DMS_Reset_DataSource_Form',
  DMS_Submit_DataSource_Form = 'DMS_Submit_DataSource_Form',
  DMS_Refresh_Member_List = 'DMS_REFRESH_MEMBER_LIST',
  DMS_Refresh_Member_Group_List = 'DMS_Refresh_Member_Group_List',
  DMS_Refresh_Project_List = 'DMS_REFRESH_PROJECT_LIST',
  DMS_SYNC_TASK_RESET_FORM = 'DMS_SYNC_TASK_RESET_FORM',
  Update_Recently_Opened_Projects = 'Update_Recently_Opened_Projects',
  Update_Current_Project_ID = 'Update_Current_Project_ID',
  DMS_Refresh_License_List = 'DMS_REFRESH_LICENSE_LIST'
}

export default EmitterKey;
