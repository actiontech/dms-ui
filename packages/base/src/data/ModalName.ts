export enum ModalName {
  // userCenter manage
  DMS_Add_Role = 'DMS_ADD_ROLE',
  DMS_Update_Role = 'DMS_UPDATE_ROLE',
  DMS_Clone_Role = 'DMS_CLONE_ROLE',

  DMS_Add_User = 'DMS_ADD_USER',
  DMS_Update_User = 'DMS_UPDATE_USER',

  DMS_Add_User_Group = 'DMS_ADD_USER_GROUP',
  DMS_Update_User_Group = 'DMS_UPDATE_USER_GROUP',

  // member manage
  DMS_Add_Member = 'DMS_ADD_MEMBER',
  DMS_Update_Member = 'DMS_UPDATE_MEMBER',
  DMS_Add_Member_Group = 'DMS_Add_Member_Group',
  DMS_Update_Member_Group = 'DMS_Update_Member_Group',
  DMS_Manage_Member_Group = 'DMS_Manage_Member_Group',

  // project manage
  DMS_Add_Project = 'DMS_ADD_PROJECT',
  DMS_Update_Project = 'DMS_UPDATE_PROJECT',

  // system config
  DMS_Import_License = 'DMS_IMPORT_LICENSE',

  // header
  Company_Notice = 'Company_Notice',

  //data export
  DMS_UPDATE_EXPORT_TASK_INFO = 'DMS_UPDATE_EXPORT_TASK_INFO',

  Cloud_Beaver_Sql_Operation_Audit_Detail = 'Cloud_Beaver_Sql_Operation_Audit_Detail',

  DMS_Create_Availability_zone = 'DMS_Create_Availability_zone',
  DMS_Update_Availability_zone = 'DMS_Update_Availability_zone'
}
