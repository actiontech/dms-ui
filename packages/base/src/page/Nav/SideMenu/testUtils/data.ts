export const mockRecentlyOpenedProjectsData = {
  recentlyProjects: [
    {
      project_id: '1',
      project_name: 'test_project'
    }
  ],
  updateRecentlyProject: jest.fn(),
  currentProjectID: '0'
};
