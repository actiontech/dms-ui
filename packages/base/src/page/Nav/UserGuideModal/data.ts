import { GetUserSystemEnum } from '@actiontech/shared/lib/api/base/service/common.enum';
import { t } from '../../../locale';

export const systemDictionary = {
  [GetUserSystemEnum.WORKBENCH]: t('dmsMenu.userGuide.sqlWorkbench.label'),
  [GetUserSystemEnum.MANAGEMENT]: t('dmsMenu.userGuide.adminPanel.label')
};
