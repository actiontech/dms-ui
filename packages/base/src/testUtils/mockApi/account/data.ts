export const mockAccessTokenInfo = {
  access_token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHA',
  is_expired: false,
  token_expired_timestamp: '2024-03-31T10:25:10+08:00'
};
