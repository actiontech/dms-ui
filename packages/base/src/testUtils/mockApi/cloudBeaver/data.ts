export const disableSqlQueryUrlData = {
  enable_sql_query: false,
  sql_query_root_uri: '/sql_query'
};

export const enableSqlQueryUrlData = {
  enable_sql_query: true,
  sql_query_root_uri: '/new_sql_query'
};
