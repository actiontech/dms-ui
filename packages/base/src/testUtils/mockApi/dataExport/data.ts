import {
  IAddDataExportTaskReply,
  IAddDataExportWorkflowReply,
  IListDataExportWorkflow
} from '@actiontech/shared/lib/api/base/service/common';
import { ListDataExportWorkflowStatusEnum } from '@actiontech/shared/lib/api/base/service/common.enum';

export const DataExportWorkflowList: IListDataExportWorkflow[] = [
  {
    project_uid: '700300',
    workflow_uid: '1752172808714063872',
    workflow_name: 'mysql-1_20240130113114',
    desc: 'desc',
    creater: {
      uid: '700200',
      name: 'admin'
    },
    created_at: '2024-01-30T11:32:06.053+08:00',
    exported_at: '2024-01-30T11:33:06.053+08:00',
    status: ListDataExportWorkflowStatusEnum.finish,
    current_step_assignee_user_list: [],
    current_step_type: ''
  },
  {
    project_uid: '700300',
    workflow_uid: '1752172808714063871',
    workflow_name: 'mysql-2_20240130113114',
    desc: 'desc',
    creater: {
      uid: '700200',
      name: 'admin'
    },
    created_at: '2024-01-30T11:32:06.053+08:00',
    exported_at: '2024-01-30T11:33:06.053+08:00',
    status: ListDataExportWorkflowStatusEnum.wait_for_approve,
    current_step_assignee_user_list: [
      {
        uid: '700200',
        name: 'admin'
      }
    ],
    current_step_type: ''
  }
];

export const AddDataExportTaskResponseData: IAddDataExportTaskReply['data'] = {
  data_export_task_uids: ['1752522203951271936']
};

export const AddDataExportWorkflowResponseData: IAddDataExportWorkflowReply['data'] =
  {
    export_data_workflow_uid: '1483489234842342'
  };
