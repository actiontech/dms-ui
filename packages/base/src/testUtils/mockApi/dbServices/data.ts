import {
  ICheckDBServiceIsConnectableReplyItem,
  IListDBService,
  IListDBServiceTipItem,
  IListGlobalDBService
} from '@actiontech/shared/lib/api/base/service/common';
import { SQLQueryConfigAllowQueryWhenLessThanAuditLevelEnum } from '@actiontech/shared/lib/api/base/service/common.enum';

export const dbServices: IListDBService[] = [
  {
    business: 'business',
    db_type: 'MySQL',
    host: '127.0.0.1',
    maintenance_times: [],
    name: 'test',
    password: 'root',
    port: '3306',
    uid: '123123',
    user: 'root',
    sqle_config: {
      rule_template_id: '1',
      rule_template_name: 'rule_template_name1',
      sql_query_config: {
        allow_query_when_less_than_audit_level:
          SQLQueryConfigAllowQueryWhenLessThanAuditLevelEnum.notice,
        audit_enabled: true
      }
    },
    instance_audit_plan_id: 1232,
    audit_plan_types: [
      { type: 'mysql_slow_log', desc: '慢日志', audit_plan_id: 1 }
    ]
  },
  {
    business: 'business',
    db_type: 'MySQL',
    host: 'localhost',
    maintenance_times: [],
    name: 'test2',
    password: '123',
    port: '3306',
    uid: '300123',
    user: 'test1'
  }
];

export const dbServicesTips: IListDBServiceTipItem[] = [
  {
    db_type: 'MySQL',
    host: '127.0.0.1',
    name: 'test',
    port: '3306',
    id: '123123'
  },
  {
    db_type: 'MySQL',
    host: 'localhost',
    name: 'test2',
    port: '3306',
    id: '300123'
  }
];

export const checkConnectableReply: ICheckDBServiceIsConnectableReplyItem[] = [
  {
    component: 'sqle',
    is_connectable: true
  },
  {
    component: 'provision',
    is_connectable: false
  }
];

export const globalDataSourceMockData: IListGlobalDBService[] = [
  {
    uid: '123456',
    name: 'vm1-mysql',
    db_type: 'MySQL',
    host: '10.186.62.41',
    port: '3306',
    business: 'vm1',
    maintenance_times: [
      {
        maintenance_start_time: {
          hour: 2,
          minute: 0
        },
        maintenance_stop_time: {
          hour: 8,
          minute: 0
        }
      }
    ],
    desc: '222',
    source: 'SQLE',
    project_uid: '700300',
    is_enable_masking: false,
    project_name: 'default',
    unfinished_workflow_num: 1,
    is_enable_audit: true
  },
  {
    uid: '123457',
    name: 'vm1-mysql',
    db_type: 'MySQL',
    host: '',
    port: '3306',
    business: 'test',
    maintenance_times: [],
    desc: '',
    source: 'SQLE',
    project_uid: '1811294305910788096',
    is_enable_masking: false,
    project_name: 'project1',
    unfinished_workflow_num: 0,
    is_enable_audit: false
  },
  {
    uid: '123458',
    name: 'vm1-oracle',
    db_type: 'Oracle',
    host: '10.186.62.41',
    port: '',
    business: 'vm1',
    maintenance_times: [],
    desc: '',
    source: 'SQLE',
    project_uid: '700300',
    is_enable_masking: false,
    project_name: 'default',
    unfinished_workflow_num: 0,
    is_enable_audit: false
  }
];

export const globalDBServicesTipsMockData = {
  db_type: ['MySQL']
};
