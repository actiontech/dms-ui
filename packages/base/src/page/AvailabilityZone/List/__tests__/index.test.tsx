import { cleanup, screen, act, fireEvent } from '@testing-library/react';
import { superRender } from '@actiontech/shared/lib/testUtil/superRender';
import AvailabilityZoneList from '../index';
import { queryBySelector } from '@actiontech/shared/lib/testUtil/customQuery';
import gateway from '../../../../testUtils/mockApi/gateway';
import { mockGatewayListData } from '../../../../testUtils/mockApi/gateway/data';
import { useDispatch, useSelector } from 'react-redux';
import { ModalName } from '../../../../data/ModalName';
import EventEmitter from '../../../../utils/EventEmitter';
import EmitterKey from '../../../../data/EmitterKey';
import { mockUseCurrentUser } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentUser';
import {
  createSpyErrorResponse,
  createSpySuccessResponse
} from '@actiontech/shared/lib/testUtil/mockApi';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useDispatch: jest.fn(),
    useSelector: jest.fn()
  };
});

describe('base/AvailabilityZoneList', () => {
  let gatewayListSpy: jest.SpyInstance;
  let deleteGatewaySpy: jest.SpyInstance;
  const dispatchSpy = jest.fn();

  beforeEach(() => {
    jest.useFakeTimers();
    mockUseCurrentUser();
    gatewayListSpy = gateway.listGateways();
    deleteGatewaySpy = gateway.deleteGateway();
    (useDispatch as jest.Mock).mockImplementation(() => dispatchSpy);
    (useSelector as jest.Mock).mockImplementation((selector) => {
      return selector({
        availabilityZone: {
          modalStatus: {
            [ModalName.DMS_Create_Availability_zone]: false,
            [ModalName.DMS_Update_Availability_zone]: false
          }
        },
        permission: {
          moduleFeatureSupport: {},
          userOperationPermissions: null
        }
      });
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
    cleanup();
  });

  it('render availability zone table', async () => {
    const { baseElement } = superRender(<AvailabilityZoneList />);
    await act(async () => jest.advanceTimersByTime(3000));
    expect(gatewayListSpy).toHaveBeenCalledTimes(1);
    expect(baseElement).toMatchSnapshot();
    expect(screen.getByText('test')).toBeInTheDocument();
    expect(screen.getByText('test2')).toBeInTheDocument();
    expect(
      screen.getByText(`共 ${mockGatewayListData.length} 条数据`)
    ).toBeInTheDocument();
    expect(screen.getAllByText('删 除')).toHaveLength(2);
    expect(screen.getAllByText('编 辑')).toHaveLength(2);
  });

  it('should render empty tips when request not success', async () => {
    gatewayListSpy.mockClear();
    gatewayListSpy.mockImplementation(() =>
      createSpyErrorResponse({ data: [] })
    );
    const { baseElement } = superRender(<AvailabilityZoneList />);
    await act(async () => jest.advanceTimersByTime(3000));
    expect(baseElement).toMatchSnapshot();
    const element = queryBySelector('.ant-table-placeholder', baseElement);
    expect(element).toBeInTheDocument();
  });

  it('should refresh availability zone table when emit "Refresh_Availability_Zone_Page" event', async () => {
    superRender(<AvailabilityZoneList />);
    await act(async () => jest.advanceTimersByTime(3000));
    expect(gatewayListSpy).toHaveBeenCalledTimes(1);
    await act(async () =>
      EventEmitter.emit(EmitterKey.Refresh_Availability_Zone_Page)
    );
    await act(async () => jest.advanceTimersByTime(3000));
    expect(gatewayListSpy).toHaveBeenCalledTimes(2);
  });

  it('should send delete availability zone request', async () => {
    gatewayListSpy.mockClear();
    gatewayListSpy.mockImplementation(() =>
      createSpySuccessResponse({
        data: [mockGatewayListData[0]],
        total_nums: 1
      })
    );
    const zoneName = mockGatewayListData[0].gateway_name;

    superRender(<AvailabilityZoneList />);
    await act(async () => jest.advanceTimersByTime(3000));
    expect(gatewayListSpy).toHaveBeenCalledTimes(1);
    fireEvent.click(screen.getByText('删 除'));
    await act(async () => jest.advanceTimersByTime(300));
    expect(
      screen.getByText(`确定要删除可用区:${zoneName}?`)
    ).toBeInTheDocument();
    fireEvent.click(screen.getByText('确 认'));
    await act(async () => jest.advanceTimersByTime(3300));
    expect(deleteGatewaySpy).toHaveBeenCalledTimes(1);
    expect(deleteGatewaySpy).toHaveBeenCalledWith({
      gateway_id: mockGatewayListData[0].gateway_id
    });
    expect(screen.getByText('删除成功')).toBeInTheDocument();
    await act(async () => jest.advanceTimersByTime(3000));
    expect(gatewayListSpy).toHaveBeenCalled();
  });

  it('should dispatch action when edit availability zone', async () => {
    gatewayListSpy.mockClear();
    gatewayListSpy.mockImplementation(() =>
      createSpySuccessResponse({
        data: [mockGatewayListData[0]],
        total_nums: 1
      })
    );
    superRender(<AvailabilityZoneList />);
    await act(async () => jest.advanceTimersByTime(3000));
    expect(gatewayListSpy).toHaveBeenCalledTimes(1);
    fireEvent.click(screen.getByText('编 辑'));
    await act(async () => jest.advanceTimersByTime(300));
    expect(dispatchSpy).toHaveBeenCalledTimes(3);
    expect(dispatchSpy).toHaveBeenNthCalledWith(2, {
      type: 'availabilityZone/updateModalStatus',
      payload: {
        modalName: ModalName.DMS_Update_Availability_zone,
        status: true
      }
    });
    expect(dispatchSpy).toHaveBeenNthCalledWith(3, {
      type: 'availabilityZone/updateSelectAvailabilityZone',
      payload: {
        availabilityZone: mockGatewayListData[0]
      }
    });
  });

  it('should dispatch action when create new availability zone', async () => {
    superRender(<AvailabilityZoneList />);
    await act(async () => jest.advanceTimersByTime(3000));
    fireEvent.click(screen.getByText('新建可用区'));
    expect(dispatchSpy).toHaveBeenCalledWith({
      type: 'availabilityZone/updateModalStatus',
      payload: {
        modalName: ModalName.DMS_Create_Availability_zone,
        status: true
      }
    });
  });
});
