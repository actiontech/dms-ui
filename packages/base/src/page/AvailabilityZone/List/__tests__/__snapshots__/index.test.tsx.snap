// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base/AvailabilityZoneList render availability zone table 1`] = `
<body>
  <div>
    <div
      class="actiontech-page-header-namespace css-1xspr0w"
    >
      <div
        class="title"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            可用区配置
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-default ant-btn-sm ant-btn-icon-only basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  class="custom-icon custom-icon-refresh"
                  height="14"
                  viewBox="0 0 14 14"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.187 2.586a5.833 5.833 0 0 1 8.59 7.762L9.918 7h1.75a4.667 4.667 0 0 0-7.899-3.367zm7.626 8.828a5.833 5.833 0 0 1-8.59-7.762L4.083 7h-1.75a4.667 4.667 0 0 0 7.899 3.367z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="extra"
      >
        <button
          class="ant-btn ant-btn-primary basic-button-wrapper css-geipcv"
          type="button"
        >
          <span>
            新建可用区
          </span>
        </button>
      </div>
    </div>
    <div
      class="ant-table-wrapper actiontech-table-namespace css-fxd6ka"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table ant-table-fixed-column ant-table-scroll-horizontal ant-table-has-fix-right"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
                style="overflow-x: auto; overflow-y: hidden;"
              >
                <table
                  style="min-width: 100%; table-layout: auto;"
                >
                  <colgroup>
                    <col />
                    <col />
                    <col
                      style="width: 164px;"
                    />
                  </colgroup>
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        可用区名称
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        服务节点
                      </th>
                      <th
                        class="ant-table-cell actiontech-table-actions-column ant-table-cell-fix-right ant-table-cell-fix-right-first"
                        scope="col"
                        style="position: sticky; right: 0px;"
                      >
                        操作
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      aria-hidden="true"
                      class="ant-table-measure-row"
                      style="height: 0px; font-size: 0px;"
                    >
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="1"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        test
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        127.0.0.1
                      </td>
                      <td
                        class="ant-table-cell actiontech-table-actions-column ant-table-cell-fix-right ant-table-cell-fix-right-first"
                        style="position: sticky; right: 0px;"
                      >
                        <div
                          class="css-new95e"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 8px;"
                            >
                              <button
                                class="ant-btn ant-btn-default ant-btn-sm actiontech-table-actions-button basic-button-wrapper css-geipcv"
                                type="button"
                              >
                                <span>
                                  编 辑
                                </span>
                              </button>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <button
                                class="ant-btn ant-btn-default ant-btn-sm ant-btn-dangerous actiontech-table-actions-button basic-button-wrapper css-geipcv"
                                type="button"
                              >
                                <span>
                                  删 除
                                </span>
                              </button>
                            </div>
                          </div>
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="2"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        test2
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        127.0.0.2
                      </td>
                      <td
                        class="ant-table-cell actiontech-table-actions-column ant-table-cell-fix-right ant-table-cell-fix-right-first"
                        style="position: sticky; right: 0px;"
                      >
                        <div
                          class="css-new95e"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 8px;"
                            >
                              <button
                                class="ant-btn ant-btn-default ant-btn-sm actiontech-table-actions-button basic-button-wrapper css-geipcv"
                                type="button"
                              >
                                <span>
                                  编 辑
                                </span>
                              </button>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <button
                                class="ant-btn ant-btn-default ant-btn-sm ant-btn-dangerous actiontech-table-actions-button basic-button-wrapper css-geipcv"
                                type="button"
                              >
                                <span>
                                  删 除
                                </span>
                              </button>
                            </div>
                          </div>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <ul
            class="ant-pagination ant-table-pagination ant-table-pagination-right actiontech-table-pagination actiontech-table-pagination-fixed"
          >
            <li
              class="ant-pagination-total-text"
            >
              <span>
                共 2 条数据
              </span>
            </li>
            <li
              aria-disabled="true"
              class="ant-pagination-prev ant-pagination-disabled"
              title="上一页"
            >
              <button
                class="ant-pagination-item-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="left"
                  class="anticon anticon-left"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="left"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
              tabindex="0"
              title="1"
            >
              <a
                rel="nofollow"
              >
                1
              </a>
            </li>
            <li
              aria-disabled="true"
              class="ant-pagination-next ant-pagination-disabled"
              title="下一页"
            >
              <button
                class="ant-pagination-item-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="right"
                  class="anticon anticon-right"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="right"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="ant-pagination-options"
            >
              <div
                aria-label="页码"
                class="ant-select ant-pagination-options-size-changer ant-select-single ant-select-show-arrow ant-select-show-search"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-label="页码"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      role="combobox"
                      type="search"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                    title="20 条/页"
                  >
                    20 条/页
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`base/AvailabilityZoneList should render empty tips when request not success 1`] = `
<body>
  <div>
    <div
      class="actiontech-page-header-namespace css-1xspr0w"
    >
      <div
        class="title"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            可用区配置
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-default ant-btn-sm ant-btn-icon-only basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  class="custom-icon custom-icon-refresh"
                  height="14"
                  viewBox="0 0 14 14"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.187 2.586a5.833 5.833 0 0 1 8.59 7.762L9.918 7h1.75a4.667 4.667 0 0 0-7.899-3.367zm7.626 8.828a5.833 5.833 0 0 1-8.59-7.762L4.083 7h-1.75a4.667 4.667 0 0 0 7.899 3.367z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="extra"
      >
        <button
          class="ant-btn ant-btn-primary basic-button-wrapper css-geipcv"
          type="button"
        >
          <span>
            新建可用区
          </span>
        </button>
      </div>
    </div>
    <div
      class="ant-table-wrapper actiontech-table-namespace css-fxd6ka"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table ant-table-empty ant-table-fixed-column ant-table-scroll-horizontal ant-table-has-fix-right"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
                style="overflow-x: auto; overflow-y: hidden;"
              >
                <table
                  style="min-width: 100%; table-layout: auto;"
                >
                  <colgroup>
                    <col />
                    <col />
                    <col
                      style="width: 164px;"
                    />
                  </colgroup>
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        可用区名称
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        服务节点
                      </th>
                      <th
                        class="ant-table-cell actiontech-table-actions-column ant-table-cell-fix-right ant-table-cell-fix-right-first"
                        scope="col"
                        style="position: sticky; right: 0px;"
                      >
                        操作
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      aria-hidden="true"
                      class="ant-table-measure-row"
                      style="height: 0px; font-size: 0px;"
                    >
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="ant-table-placeholder"
                    >
                      <td
                        class="ant-table-cell"
                        colspan="3"
                      >
                        <div
                          class="ant-table-expanded-row-fixed"
                          style="width: 0px; position: sticky; left: 0px; overflow: hidden;"
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
